BPF Tests: panic in case of wrong naming

In the BPF tests, when the CHECK doesn't have the same name as its function or SETUP/PKTGEN, the tests fail in a non-obvious way:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]                           
	panic: runtime error: invalid memory address or nil pointer dereference                                       
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x2da3fe8]                                      
                                                                                                              
goroutine 122 [running]:                                                                                      
testing.tRunner.func1.2({0x31bd7a0, 0x608b1e0})                                                               
	/usr/local/go/src/testing/testing.go:1734 +0x21c                                                              
testing.tRunner.func1()                                                                                       
	/usr/local/go/src/testing/testing.go:1737 +0x35e                                                              
panic({0x31bd7a0?, 0x608b1e0?})                                                                               
	/usr/local/go/src/runtime/panic.go:792 +0x132                                                                 
github.com/cilium/cilium/bpf/tests/bpftest.loadAndRunSpec(0xc000103880, {0x3cee7e0, 0xc0009749c0}, {0x0, 0x0})
	/go/src/github.com/cilium/cilium/bpf/tests/bpftest/bpf_test.go:202 +0x908                                     
github.com/cilium/cilium/bpf/tests/bpftest.TestBPF.func1(0xc000103880?)                                       
	/go/src/github.com/cilium/cilium/bpf/tests/bpftest/bpf_test.go:98 +0x33                                       
testing.tRunner(0xc000103880, 0xc00112b2f0)                                                                   
	/usr/local/go/src/testing/testing.go:1792 +0xf4                                                               
created by testing.(*T).Run in goroutine 35                                                                   
	/usr/local/go/src/testing/testing.go:1851 +0x413                                                              
FAIL	github.com/cilium/cilium/bpf/tests/bpftest	4.545s  
```
We could probably improve that. The name I changed was `ipsec_redirect_bad_identities6` -> `ipsec_redirect_bad_identitie6`.

cc @dylandreimerink

--- a/bpf/tests/bpftest/bpf_test.go
+++ b/bpf/tests/bpftest/bpf_test.go
@@ -175,6 +175,21 @@ func loadAndRunSpec(t *testing.T, entry fs.DirEntry, instrLog io.Writer) []*cove
 
 	testNameToPrograms := make(map[string]programSet)
 
+	checkProgExists := func(progName, testName, progType string) {
+		checkProgName := strings.Replace(progName, progType, "check", 1)
+		if spec, ok := spec.Programs[checkProgName]; ok {
+			match := checkProgRegex.FindStringSubmatch(spec.SectionName)
+			if match[1] != testName {
+				t.Fatalf(
+					"File '%s' contains a %s program for '%s' test, but no check program.",
+					elfPath,
+					progType,
+					testName,
+				)
+			}
+		}
+	}
+
 	for progName, spec := range spec.Programs {
 		match := checkProgRegex.FindStringSubmatch(spec.SectionName)
 		if len(match) == 0 {
@@ -183,25 +198,17 @@ func loadAndRunSpec(t *testing.T, entry fs.DirEntry, instrLog io.Writer) []*cove
 
 		progs := testNameToPrograms[match[1]]
 		if match[2] == "pktgen" {
+			checkProgExists(progName, match[1], match[2])
 			progs.pktgenProg = coll.Programs[progName]
 		}
 		if match[2] == "setup" {
+			checkProgExists(progName, match[1], match[2])
 			progs.setupProg = coll.Programs[progName]
 		}
 		if match[2] == "check" {
 			progs.checkProg = coll.Programs[progName]
 		}
 		testNameToPrograms[match[1]] = progs
-	}
-
-	for progName, set := range testNameToPrograms {
-		if set.checkProg == nil {
-			t.Fatalf(
-				"File '%s' contains a setup program in section '%s' but no check program.",
-				elfPath,
-				spec.Programs[progName].SectionName,
-			)
-		}
 	}
 
 	// Collect debug events and add them as logs of the main test

